core/display.js

```diff
@@ -405,12 +297,21 @@
                     'height': h,
                 });
             } else {
-                var x1 = old_x - this._viewportLoc.x;
-                var y1 = old_y - this._viewportLoc.y;
-                var x2 = new_x - this._viewportLoc.x;
-                var y2 = new_y - this._viewportLoc.y;
+                // Due to this bug among others [1] we need to disable the image-smoothing to
+                // avoid getting a blur effect when copying data.
+                //
+                // 1. https://bugzilla.mozilla.org/show_bug.cgi?id=1194719
+                //
+                // We need to set these every time since all properties are reset
+                // when the the size is changed
```

Comment:
In theory, yes. However I couldn't see any issue so I left it out. It may be that the problem is very small and only noticeable when you keep copying the same data many times.